### DISCUSSION TITLE
Issues and clarification with generating propeller geometry from a CAD file using the generate_rotor method

### DISCUSSION THREAD

**Discussion by jayjiejie** [byuflowlab/FLOWUnsteady#162] [Q&A]
Dear FLOWUnsteady team and everyone

I have been trying to use the [Generating components](https://flow.byu.edu/FLOWUnsteady/api/flowunsteady-vehicle-components/#Rotor) method to create the INSEAN E779A propeller geometry in FLOWUnsteady.

I tried to play around with the geometrical files by extracting the relevant geometrical feature quantities from the CAD file, e.g. chorddist.csv, weistdist.csv, sweepdist.csv, heightdist.csv and airfoils.csv and my blade profile in FLOWUnsteady is not the same as the CAD file. 

I usually align the propeller blade (CAD file) vertically up and taking a series horizontal slices at different vertical distance to get the sectional profiles. However, I find myself getting a bit confused with the followings terminology definitions:

1. **twistdist vs. pitchdist**
According to the web page for Generating components - Rotor [https://flow.byu.edu/FLOWUnsteady/api/flowunsteady-vehicle-components/#Rotor](url), it uses twistdist.csv. However, the /examples/propeller tutorial uses apc10x7_pitchdist.csv (located in /database/rotors directory) and its content still use the column name of twist inside. Does it matter if twistdist or pitchdist is used in the blade.csv file? My understanding (which could be wrong) are the **twist** of the propeller blade is the angle between the profile chord of the blade tip and the profile chord of the blade root and the **pitch** is the angle between the chord profile and the plane of rotation. Hence, the two definitions are different.

2. **sweepdist**
For sweepdist, the definition (found in the FLOWUnsteady) is y/R and y distance of LE from the middle point of hub. Would you be able to clarify the definition for the **middle point of hub**?

3. **heightdist**
This is a similar question as the item 2 above. The heightdist is defined (in the FLOWUnsteady) as z/R height of leading edge from top face of hub. How do we quantify this parameter from the CAD file? Which part of the leading edge we should take as the measurement point? Where is the top surface of hub?

4. **airfoils.csv**
 For all the aero files in the airfoil.csv, presumably they are the airfoil profiles normalised with  the chrod length (c becomes 1), LE at c = 0 and TE at c = 1, AoA = 0 deg. Does it matter if the points start from LE (TE), or from the top to bottom surface (bottom to top surface)?

It looks like there are a few of us trying to create the propeller geometry from a CAD file here but ended up with various issues. I feel it is worth spending time to get through those definitions used in each of the csv files so that we are not making silly mistakes when generating the geometrical files from the CAD. 

I appreciate any suggestions on how to correctly create the propeller model from the CAD files. Thank you very much.

Jay

**Comment by EdoAlvarezR** [comment]
Hi Jay!

1. **twistdist vs. pitchdist:** In early versions of FLOWUnsteady we used to call it pitch, but later we started to call it twist. Sorry for the confusion. What we mean is the angle between the chord profile and the plane of rotation when the collective pitch is zero (`uns.generate_rotor(...; pitch=0.0, ...)`).
2. **sweepdist:** By middle point of the hub we mean the axis of rotation. Hopefully the following diagram of `y/R` is self explanatory:
![image](https://github.com/byuflowlab/FLOWUnsteady/assets/14857108/a170ee6b-fe39-4beb-82b1-f6c08815550f)
See the rest of the slides in the docs [here](https://flow.byu.edu/FLOWUnsteady/api/flowunsteady-vehicle-components/#Rotor)
3. **heightdist:** `Which part of the leading edge we should take as the measurement point?` I typically select what would be the leading-edge point if I untwisted the airfoil cross section. `Where is the top surface of hub?` It's completely arbitrary. The exact z/R position doesn't really matters; what matters is the distribution itself (meaning, how z changes along the blade).
![image](https://github.com/byuflowlab/FLOWUnsteady/assets/14857108/f8392995-2e7c-42c7-a1f7-889c34c2e825)
4. **airfoils.csv:** Yes, LE at `x/c=0` and TE at `x/c=1`. Ideally it would be the untwisted airfoil (`AOA=0`), but if you happen to give it a twisted angle, that just means that the twist compounds with whathever you give it in `twistdist`. We recommend sorting the airfoil contour to start at the TE, go over the top to the LE, and go back to the TE from the bottom surface. It probably won't through off the simulation if you don't follow that ordering though.

Hopefully this helps, but feel free to ask more questions!

**Reply by EdoAlvarezR** [reply]
Related to #158

**Reply by jayjiejie** [reply]
Hi Edo

Thank you again for your previous reply and I managed to re-generate the blade geometry> However, I am keep getting the Mach correction error as [#185](https://github.com/byuflowlab/FLOWUnsteady/issues/185): 

ERROR: LoadError: Mach correction requested on Ma = 4.468115058306802 >= 1.0

I did visualise the flow filed and the simulation was blown up indeed. I tried to tweak the simulation parameters without too much success. I am wondering if you have any suggestions for approaching this Mach correction error problem?

Thank you very much.

Kind regards,
Jay
